driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/MongoCollection.java

```diff
@@ -156,6 +188,22 @@ public interface MongoCollection<TDocument> {
      */
     MongoCollection<TDocument> withReadConcern(ReadConcern readConcern);
 
+    /**
+     * Create a new MongoCollection instance with the set time limit for the full execution of an operation.
+     *
+     * <ul>
+     *   <li>{@code 0} means infinite timeout.</li>
+     *    <li>{@code > 0} The time limit to use for the full execution of an operation.</li>
+     * </ul>
+     *
+     * @param timeout the timeout, which must be greater than or equal to 0
+     * @param timeUnit the time unit
+     * @return a new MongoCollection instance with the set time limit for the full execution of an operation
+     * @since CSOT
+     * @see #getTimeout
+     */
+    MongoCollection<TDocument> withTimeout(long timeout, TimeUnit timeUnit);
```

Comment:
See https://github.com/mongodb/mongo-java-driver/pull/1349#discussion_r1538941852 - clarify regarding use of times below 1ms.